hugolib/page.go

```diff
@@ -1070,13 +1070,44 @@ func (p *Page) update(f interface{}) error {
 	p.Params["draft"] = p.Draft
 
 	if p.Date.IsZero() && p.s.Cfg.GetBool("useModTimeAsFallback") {
+
 		fi, err := p.s.Fs.Source.Stat(filepath.Join(p.s.PathSpec.AbsPathify(p.s.Cfg.GetString("contentDir")), p.File.Path()))
 		if err == nil {
+			p.s.Log.DEBUG.Printf("using file modification time as fallback for page %s", p.File.Path())
 			p.Date = fi.ModTime()
 			p.Params["date"] = p.Date
 		}
 	}
 
+	if p.Date.IsZero() && p.s.Cfg.GetString("useFilenameDateAsFallback") != "" {
```

Comment:
fixed use of bool and I changed the default to `true` including updating the docs/commit messages to match.